### DISCUSSION TITLE
firestore timestamp

### DISCUSSION THREAD

**Discussion by bobby-cheema** [invertase/react-native-firebase#5430] [Q&A]
Hi all ,

 i am very new to react-native and firestore. I  am storing starttime and endtime from my react-native app to firestore . I can get the data from firestore and save it to my state and can render it on my screen , however i could not figure out how to get(). and store date  and time from firestore to mystate which i am getting as nested object. any example code  would really be appreciated . I have created the following mess so far . 

```javascript
const [mydata, setmydata] = useState([]);
useEffect(() => {
    const jobs = [];
    firestore()
      .collection('times')
      .get()
      .then(folder => {
        folder.forEach(file => {
          // console.log(file.data());
          jobs.push({
            ...file.data(),
            key: file.id,
          });
          console.log('key', file.id);
          setmydata(jobs);
        });
      })
      .catch(e => {
        console.log('error is', e);
      });
  }, []);

  return (
    <View style={styles.container}>
      <Appbar.Header>
        <Appbar.Content title="Jobs Added" subtitle="Jobs done so far" />
      </Appbar.Header>

      <Text style={styles.text}>Jobs done </Text>
      <FlatList
        data={mydata}
        renderItem={({item}) => {
          const startdate = new Date(item.starwork.nanoseconds.toString);
          return <Text>{startdate} </Text>;
        }}
      />
    </View>
  );
};  
```

**Comment by mikehardy** [comment]
Hi there! This is not really react-native-firebase specific, it's just general javascript / programming - I suggest stackoverflow as a much superior place for questions like these
